core/src/main/java/com/sequenceiq/distrox/v1/distrox/converter/InstanceTemplateParameterConverter.java

```diff
@@ -68,10 +69,11 @@ private AwsEncryptionV4Parameters convert(AwsEncryptionV1Parameters source, Deta
             if (environmentEncryptionKeyArn != null && !environmentEncryptionKeyArn.isEmpty()) {
                 response.setKey(environmentEncryptionKeyArn);
                 response.setType(EncryptionType.CUSTOM);
-            } else {
-                response.setKey(source.getKey());
-                response.setType(source.getType());
             }
+            if (dataHubEncryptionKey == null || dataHubEncryptionKey.isEmpty() &&
+                    environmentEncryptionKeyArn == null || environmentEncryptionKeyArn.isEmpty()) {
```

Comment:
`&&` has a higher precedence than `||` so this condition does not express what you intended. It should likely be reformulated as
```
            if ((dataHubEncryptionKey == null || dataHubEncryptionKey.isEmpty()) &&
                    (environmentEncryptionKeyArn == null || environmentEncryptionKeyArn.isEmpty())) {
```

On the other hand, I do not think this separate `if` statement is even needed. See next comment.